### PR TITLE
Add Reddit comments to datasets

### LINKED ISSUE #22482: Reddit comments dataset
Needed for #1158.

### PR DESCRIPTION
**Pull request by rfraposa**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

Closes https://github.com/ClickHouse/ClickHouse/issues/22482

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit 6a136897e375e61bf59a25c6dc65e4e2570e7641 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49968/6a136897e375e61bf59a25c6dc65e4e2570e7641/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docs/en/getting-started/example-datasets/reddit-comments.md [hunk 1]
```diff
@@ -0,0 +1,636 @@
+---
+slug: /en/getting-started/example-datasets/reddit-comments
+sidebar_label: Reddit comments
+---
+
+# Reddit comments dataset
+
+This dataset contains publicly-available comments on Reddit that go back to December, 2005, to March, 2023, and contains over 7B rows of data. The raw data is in JSON format in compressed `.zst` files and the rows look like the following:
```
Comment (alexey-milovidov) [inline docs/en/getting-started/example-datasets/reddit-comments.md:8]: This is incorrect.

1. It is not 7B rows of data.
2. It is not available in zst files. zst + bz2 + xz

Comment (alexey-milovidov) [reply docs/en/getting-started/example-datasets/reddit-comments.md:8]: Screenshot_20230520_022638

File: docs/en/getting-started/example-datasets/reddit-comments.md [hunk 2]
```diff
@@ -0,0 +1,636 @@
...
+:::
+
+2. We are going to start with one month of data, but if you want to simply insert every row - skip ahead to step 8 below. The following file has 86M records from December, 2017:
+
+```sql
+INSERT INTO reddit
+    SELECT *
+    FROM s3Cluster(
+        'default',
+        'https://clickhouse-public-datasets.s3.eu-central-1.amazonaws.com/reddit/original/RC_2017-12.xz',
```
Comment (alexey-milovidov) [inline docs/en/getting-started/example-datasets/reddit-comments.md:83]: It does not make sense to use `s3Cluster` with a single file.
The `s3Cluster` is effective only with multiple files.

File: docs/en/getting-started/example-datasets/reddit-comments.md [hunk 3]
```diff
@@ -0,0 +1,636 @@
...
+
+```sql
+INSERT INTO reddit
+SELECT *
+FROM s3Cluster(
+    'default',
+    'https://clickhouse-public-datasets.s3.amazonaws.com/reddit/original/RC*',
+    'JSONEachRow'
+    )
+SETTINGS zstd_window_log_max = 31;
```
Comment (alexey-milovidov) [inline docs/en/getting-started/example-datasets/reddit-comments.md:211]: Let's explain it right here.

File: docs/en/getting-started/example-datasets/reddit-comments.md [hunk 4]
```diff
@@ -0,0 +1,636 @@
...
+```
+
+The response looks like:
+
+```response
+0 rows in set. Elapsed: 61187.839 sec. Processed 6.74 billion rows, 2.06 TB (110.17 thousand rows/s., 33.68 MB/s.)
```
Comment (alexey-milovidov) [inline docs/en/getting-started/example-datasets/reddit-comments.md:217]: It is wrong, should be over 14 billion records.

Comment (rfraposa) [reply docs/en/getting-started/example-datasets/reddit-comments.md:217]: Resolved in https://github.com/ClickHouse/ClickHouse/pull/50419

**Comment by rfraposa** [comment]
Fixes made in https://github.com/ClickHouse/ClickHouse/pull/50419

**Pull request merged**